### PR TITLE
Add PHP implementation of the server

### PR DESCRIPTION
**Pull request by youssef-stripe**

### CODE REVIEW

File: server/php/index.php [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  }
+);
+
+// General config
+$app->get('/config', function (Request $request, Response $response, array $args) {
+  $config = $this->get('settings')['stripe'];
+  return $response->withJson([
+    'stripePublishableKey' => $config['publishableKey'],
+    'stripeCountry' => $config['accountCountry'],
+    'country' => $config['accountCountry'],
```
Comment (thorsten-stripe) [inline server/php/index.php]: this is meant to be the default country of the payment form, and could therefore be different from your Stripe account's (your business ) country.

File: server/php/settings.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          'staticDir' => __DIR__ . '/../../public/',
+
+          // Find them at https://dashboard.stripe.com/account/apikeys
+          'publishableKey' => '',
+          'secretKey' => '',
+
+          // Find it at https://dashboard.stripe.com/account/webhooks
+          'webhookSecret' => '',
+
+          // Your prefered country and currency
```
Comment (thorsten-stripe) [inline server/php/settings.php]: not necessarily preferred, but rather the country of your Stripe account.

File: server/php/settings.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          'accountCountry' => 'US',
+          'accountCurrency' => 'usd',
```
Comment (thorsten-stripe) [inline server/php/settings.php]: not necessarily the account currency, but rather the presentment currency.

File: server/php/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+cd server/php
+composer install
+```
+
+Then update your `settings.php` file with your own [Stripe API keys](https://dashboard.stripe.com/account/apikeys) and any other configuration details you might want to add.
+
+That's it, you can now run the application by using the PHP built-in server still from the `server/php` directory:
+
+```
+php -S localhost:8888 -t ./ index.php
```
Comment (thorsten-stripe) [inline server/php/README.md]: seeing that you have it in the composer script, maybe change to `composer start`?

File: server/php/README.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+### Testing Webhooks
+
+If you want to test [receiving webhooks](https://stripe.com/docs/webhooks), we recommend using ngrok to expose your local server.
+
+First [download ngrok](https://ngrok.com) and start your PHP application.
+
+[Run ngrok](https://ngrok.com/docs). Assuming your PHP application is running on the port 8888, you can simply run ngrok in your Terminal in the directory where you downloaded ngrok:
+
+```
+./ngrok http 8888
```
Comment (thorsten-stripe) [inline server/php/README.md]: should remove `./` since ngrok is installed globally

**Comment by thorsten-stripe** [comment]
@youssef-stripe thanks a lot for this! Only a couple minor nits. Once fixed, please also update the master README to include a link to the PHP server folder: https://github.com/stripe/stripe-payments-demo/blob/master/README.md#getting-started-with-node

File: server/php/index.php [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+  $logger = $this->get('logger');
+
+  $event = $request->getParsedBody();
+
+  // Parse the message body (and check the signature if possible)
+  $webhookSecret = $this->get('settings')['stripe']['webhookSecret'];
+  if ($webhookSecret) {
+    try {
+      $event = \Stripe\Webhook::constructEvent(
+        @file_get_contents('php://input'),
```
Comment (karlr-stripe) [inline server/php/index.php]: I know we have to get the raw request body here(and not a parsed version that Slim may be putting into the the `$request` variable, like `getParsedBody`), or the signatures won't match. Is there maybe a more idiomatic way to do that in Slim? Would `$request->getContent()` or [similar](https://www.slimframework.com/docs/v3/objects/response.html#the-response-body) work? Feel free to ignore this, I'm just wondering if we can also use this as a nice example of how to get a raw body for a webhook in this framework.

File: server/php/settings.php [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        // Monolog settings
+        'logger' => [
+            'name' => 'slim-app',
+            'path' => isset($_ENV['docker']) ? 'php://stdout' : __DIR__ . '/logs/app.log',
+            'level' => \Monolog\Logger::DEBUG,
+        ],
+
+        'stripe' => [
+          // You shouldn't have to touch this
+          'apiVersion' => '2019-02-11',
```
Comment (karlr-stripe) [inline server/php/settings.php]: Any reason for this not to be the latest version `2019-03-14` as of right now?

File: server/php/index.php (hunk 1 above)
Comment (youssef-stripe) [reply server/php/index.php]: Settings.php updated.

File: server/php/settings.php (hunk 2 above)
Comment (youssef-stripe) [reply server/php/settings.php]: Settings.php updated.

File: server/php/settings.php (hunk 3 above)
Comment (youssef-stripe) [reply server/php/settings.php]: Settings.php updated.

File: server/php/README.md (hunk 4 above)
Comment (youssef-stripe) [reply server/php/README.md]: Done!

File: server/php/README.md (hunk 5 above)
Comment (youssef-stripe) [reply server/php/README.md]: Done!

File: server/php/index.php (hunk 6 above)
Comment (youssef-stripe) [reply server/php/index.php]: Indeed, replaced with $request->getBody()

File: server/php/settings.php (hunk 7 above)
Comment (youssef-stripe) [reply server/php/settings.php]: No reason, just bumped the version.

File: server/php/settings.php [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        'stripe' => [
+          // You shouldn't have to touch this
+          'apiVersion' => '2019-03-14',
+
+          // Update this path if you want to move your public folder
+          'staticDir' => __DIR__ . '/../../public/',
+
+          // Find them at https://dashboard.stripe.com/account/apikeys
+          'publishableKey' => '',
+          'secretKey' => '',
```
Comment (thorsten-stripe) [inline server/php/settings.php]: We shouldn't have the secret key in a file that is committed to git. Can we made take this approach? https://medium.com/@hfally/how-to-create-an-environment-variable-file-like-laravel-symphonys-env-37c20fc23e72

Comment (youssef-stripe) [reply server/php/settings.php]: Good point, added ini file, see e131cf48cf1dc9d9c3113821af789c6497c285e6

**Pull request merged**